### ISSUE TITLE
Struggling to deploy Rancher Desktop via Intune

### ISSUE THREAD

**Issue by Jiddeh** [rancher-sandbox/rancher-desktop#7356]
Hello,

Not sure if this is the right place for this or if i need to try the Slack channel...

My company are trying to roll out the deployment of Rancher Desktop and ideally we want to deploy it via our Company Portal via Intune. The MSI package has been uploaded as a line-of-business app and in an ideal world it just works as the install commands are packaged into the MSI. However it seems something isn't quite right with it and it constantly fails.

Just wondering if anyone has managed to achieve deploying this via Intune and if so, how?

Thanks,
Jack

**Comment by ademaf** [comment]
Hello Jack,

Yes I'm facing the same issue. When installing it via cmd prompt open by the user installation is working also when I use Winget for it. But when I use winget or Win32 for installation via the company portal it fails and I can find why?

**Comment by SYSMANTEAM** [comment]
We are having the same issue with deploying rancher Desktop using Intune. We have tried in System Context and user context, also allowing the user to interact with the application does not work, the application fails with Fatal error during installation (0x80070643)

Yet it works manually, i suspect it has something to do with the install asking if it should be installed in user context or System context.

makes it difficult for us, as our users do not have admin access, but we use Intune to deploy applications in System context. We would like to deploy this in system context so that users can use all the available functionalities.

Silent install parameters used (Works manually) with admin access.

msiexec /i "Rancher.Desktop.Setup.1.15.1.msi" /qn

**Comment by ademaf** [comment]
![rancherdesktop](https://github.com/user-attachments/assets/7d441bd3-e294-4c32-9c25-f88f764979a4)

We changed the install behaviour to User and now it installs with out a problem. The only thing is that you can't uninstall the application via the company portal but that's no problem because RancherDesktop has auto update and that means that after an update you can uninstall with the old MSI file.

**Comment by mook-as** [comment]
You may want to try setting the standard MSI properties `ALLUSERS=[0|1]` and `MSIINSTALLPERUSER=[0|1]` (but I'd have thought InTune would have done that already — I haven't had a need to use InTune personally so I don't know for sure).  See [docs](https://learn.microsoft.com/en-us/windows/win32/msi/msiinstallperuser) for details.

It's also possible to set `WSLINSTALLED=1` to force the Rancher Desktop installer to skip checking for WSL and assume it is installed.

**Comment by Jiddeh** [comment]
> ![rancherdesktop](https://private-user-images.githubusercontent.com/74956093/360909800-7d441bd3-e294-4c32-9c25-f88f764979a4.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3NTc3NTksIm5iZiI6MTcyNDc1NzQ1OSwicGF0aCI6Ii83NDk1NjA5My8zNjA5MDk4MDAtN2Q0NDFiZDMtZTI5NC00YzMyLTljMjUtZjg4Zjc2NDk3OWE0LkpQRz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI3VDExMTczOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFiYmIzMDVhZGVjNTc1MjUwNTczOTdkMDNmMjA5NTM3Nzg0YTI0NWRmYTlmOWUwYjMzM2RjY2E3ODk5YWEwYTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BeZniKIIcF1BCWWNMwTMqPNIEfQqO51LkbwxlZqHO4Y)
> 
> We changed the install behaviour to User and now it installs with out a problem. The only thing is that you can't uninstall the application via the company portal but that's no problem because RancherDesktop has auto update and that means that after an update you can uninstall with the old MSI file.

This doesn't seem to work for me in our setup, still just fails when trying to install from Company Portal...

**Comment by Jiddeh** [comment]
> You may want to try setting the standard MSI properties `ALLUSERS=[0|1]` and `MSIINSTALLPERUSER=[0|1]` (but I'd have thought InTune would have done that already — I haven't had a need to use InTune personally so I don't know for sure). See [docs](https://learn.microsoft.com/en-us/windows/win32/msi/msiinstallperuser) for details.
> 
> It's also possible to set `WSLINSTALLED=1` to force the Rancher Desktop installer to skip checking for WSL and assume it is installed.

Where can these properties be adjusted?

**Comment by mook-as** [comment]
They are on the `msiexec` command line, see `msiexec /?`:
> `msiexec /i "Rancher.Desktop.Setup.1.15.1.msi" /qn WSLINSTALLED=1 ALLUSERS=0`

**Comment by Jiddeh** [comment]
Tried it with the command and still no luck, just fails after the progress bar of installing hit 100%... so WSL has to be installed to even attempt the installation silently?

**Comment by jandubois** [comment]
> so WSL has to be installed to even attempt the installation silently

Yes, WSL2 is a prerequisite for installing Rancher Desktop. `WSLINSTALLED=1` is a promise that a compatible version is already installed. It is expected to break if you were not telling the truth. 😄
